src/ResourceStore.php

```diff
@@ -0,0 +1,237 @@
+<?php
+
+namespace silverorange;
+
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Client as HttpClient;
+
+class ResourceStore
+{
+    // {{{ protected properties
+
+    protected $json_api_base;
+    protected $json_api_headers;
+
+    protected $http_client;
+
+    protected $class_by_type = array();
+
+    protected $resources = array();
+
+    // }}}
+    // {{{ public function __construct()
+
+    public function __construct($json_api_base, $json_api_headers = [])
+    {
+        $this->json_api_base = $json_api_base;
+        $this->json_api_headers = $json_api_headers;
+
+        $this->initHttpClient();
+    }
+
+    // }}}
+    // {{{ protected function initHttpClient()
+
+    protected function initHttpClient()
```

Comment:
Could be renamed `initDefaultHttpClient` if it's possible to inject an existing client in the constructor